pkg/controllers/provisioning/scheduling/nodeclaim.go

```diff
@@ -103,26 +167,94 @@ func (n *NodeClaim) Add(pod *v1.Pod, podData *PodData) error {
 		return err
 	}
 
+	reservedOfferings, offeringsToRelease, err := n.reserveOfferings(remaining, nodeClaimRequirements)
```

Comment:
Can you just diff the current `n.reservedOfferings` with the offerings that are returned from this function? Maybe there's a way to store this efficiently so you aren't having to do a find across all offerings to do this but that strikes me as simpler than having to build-up and handle partitioning

Reply:
I've changed my mind, I think the particular way I was building it up wasn't necessarily more efficient than taking the diff, since it included offerings we may have never reserved, and I do agree that taking the diff is a bit cleaner. I've updated it in the latest revision (which should be pushed in the next couple of hours).